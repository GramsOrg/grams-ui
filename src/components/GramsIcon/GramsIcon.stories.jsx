import React from 'react';

import { action } from "@storybook/addon-actions";

import GramsIcon from './GramsIcon.tsx';

export default {
  title: 'General/GramsIcon',
  component: GramsIcon,
  parameters: {
    componentSubtitle: 'Component description',
    storyshots: { disable: false },
  },
  argTypes: {
    dir: {
      control: 'radio',
      options: ['ltr', 'rtl']
    },
    inverted: {
      control: 'boolean'
    }
  }
};

const Template = ({dir, inverted, ...props}) => {
  return (
    <GramsIcon
      dir={dir}
      inverted={inverted}
      {...props}
    />
  );
};

export const Default = Template.bind({});
Default.args = {
  dir: 'ltr',
  inverted: false,
  name: 'world'
}

export const RTL = Template.bind({});
RTL.args = {
  dir: 'rtl',
  inverted: false,
  name: 'user'
}
