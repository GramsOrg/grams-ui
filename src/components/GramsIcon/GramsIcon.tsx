import React, { useState, useEffect } from 'react';

import loadable from  '@loadable/component';

import {
  Icon as SemanticIcon,
  List,
  SemanticICONS,
  SemanticCOLORS
} from 'semantic-ui-react';

import SemanticValues from './SemanticValues';
import '../../main.css';
import { IconSizeProp } from 'semantic-ui-react/dist/commonjs/elements/Icon/Icon';

export interface IGramsIconProps {

  /**
   * Defines the direction of the component
   */
  dir?: 'ltr' | 'rtl';

  /**
   * Toggles between light and dark modes
   */
  inverted?: boolean;

  /**
   * Name of the icon to display.
   */
  name: string,

  /**
   * The size of the icon.
   */
  size?: IconSizeProp,

  /**
   * Whether to fallback to React icons if the Semantic UI icon is not found.
   */
  fallback?: boolean,
}

const defaultProps = {
  dir: 'ltr',
  inverted: false,
  size: 'large',
  fallback: true
};

const GramsIcon = (props: IGramsIconProps) => {

  return (
    <SemanticIcon
      name={ props.name as SemanticICONS }
      size={ props.size }
    />
  );

  /*const [tag, setTag] = useState<JSX.Element>(<div />)

	const [library, iconComponent] = props.name.split('/')

  useEffect(() => {
    import(`react-icons/${library}`)
    .then((value) => {
      const component = value[iconComponent];
      console.log(component())
      setTag(component())
    });
  }, []);

  const isSemantic: boolean = !library; //props.name in SemanticValues;

  if (isSemantic) {
    return (
      <SemanticIcon name={ props.name as SemanticICONS } />
    );
  }

  //const ReactIcons = loadable.lib(() => import(`react-icons/${library}`));
  

  //const Tag: JSX.Element = ReactIcons.then(icons => icons[iconComponent]);

  //return tag;

  /*type GetIconProps = {
    icon: keyof typeof ReactIcons;
  }

  const getIcon = ({icon}: GetIconProps) => {
    const TagName = ReactIcons[icon];
    console.log(TagName)
    return <TagName />;
  };

  console.log(ReactIcons)
  //const ReactIcon = ReactIcons[iconComponent];
  //console.log(iconComponent)
  //console.log(ReactIcon)

  return getIcon({icon: iconComponent})*/

  // return (<ReactIcons size={64} />);
};

GramsIcon.defaultProps = defaultProps;

export default GramsIcon;

// return React.lazy(() => import(`react-icons/${lib}`))

  /*const lib = props.name.slice(0, 2).toLowerCase();

  console.log(lib)

  import(`react-icons/${lib}`)
  .then(value => setParsed(value))

  if (parsed) {
    return React.lazy(() => import(`react-icons/${lib}`))
    //const Tag = props.name as keyof JSX.IntrinsicElements
    //return <Tag />;
    //return React.createElement(props.name, {});
  }

  return (
    <>
    <FcAddressBook size={64} />
    </>
  );*/