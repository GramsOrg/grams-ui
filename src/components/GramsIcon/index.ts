import GramsIcon from './GramsIcon';
export default GramsIcon;
