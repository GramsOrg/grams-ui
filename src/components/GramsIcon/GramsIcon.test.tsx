/**
 * @jest-environment jsdom
 */

import React from 'react';
import { describe, expect, test } from '@jest/globals';
import { render } from '@testing-library/react';

import GramsIcon from './GramsIcon';

describe('GramsIcon Component', () => {
  test('it should match the snapshot', () => {
    const { asFragment } = render(<GramsIcon name='world' />);
    expect(asFragment()).toMatchSnapshot();
  });
});
