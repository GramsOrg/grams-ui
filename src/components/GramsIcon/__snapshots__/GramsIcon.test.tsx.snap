// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GramsIcon Component it should match the snapshot 1`] = `
<DocumentFragment>
  <i
    aria-hidden="true"
    class="world large icon"
  />
</DocumentFragment>
`;
